hasher-matcher-actioner/hmalib/lambdas/pdq/pdq_hasher.py

```diff
@@ -70,18 +72,19 @@ def lambda_handler(event, context):
             with metrics.timer(metrics.names.pdq_hasher_lambda.hash):
                 pdq_hash, quality = pdq_hasher.pdq_from_bytes(bytes_)
 
-            hash_record = PipelinePDQHashRecord(
-                key, pdq_hash, datetime.datetime.now(), quality
+            hash_records.append(
+                PipelinePDQHashRecord(key, pdq_hash, datetime.datetime.now(), quality)
             )
 
-            hash_record.write_to_table(records_table)
-
-            # Publish to SQS queue
-            sqs_client.send_message(
-                QueueUrl=OUTPUT_QUEUE_URL,
-                MessageBody=json.dumps(hash_record.to_sqs_message()),
-            )
+    for hash_record in hash_records:
+        hash_record.write_to_table(records_table)
```

Comment:
This is probably worse for the the load numbers.

Before, for one loop, we had
```
|---hash---ddb-write---hash---ddb-write---hash---ddb-write
```

Now, we do
```
|---hash---hash---hash---ddb-write-ddb-write-ddb-write-|
```

The second one will be spikier load, right? I'm forgetting the problem we're trying to solve here? What kind of errors would reduce if we did this?

Reply:
"The main theorized advantage here is that an s3 error will not results in rewrite and re-enqueue."

So no help to dynamodb or sqs yet but if s3 throttles this helps. The next step would be to batch write but I didn't want to change write interface as part of the first pass.

Reply:
@schatten you're right, we don't have enough information to know if this will help so I will skip this change for now.